accepted/2024/support-nuget-authentication-plugins-dotnet-tools.md

```diff
@@ -0,0 +1,500 @@
+
+**Disadvantages:**
+
+- Configuring the setting per feed can be cumbersome if multiple private feeds can use the same .NET tool for authentication.
+
+An alternative approach to address this disadvantage is to declare the plugins explicitly outside of the package source sections.
+This approach allows customers to specify the dependency only once, without configuring it per private feed as discussed above.
+
+```xml
+<?xml version="1.0" encoding="utf-8"?>
+<configuration>
+  <DotNetToolPlugins>
+    <Authentication>
+      <plugin id="CredentialProvider.Microsoft" />
+      <plugin id="CredentialProvider.GitHub" />
+      <!-- Allows specifying multiple authentication plugins with additional details -->
+    </Authentication>
+  <!-- This format provides flexibility for future plugin types and detailed configurations -->
+  </DotNetToolPlugins>
+</configuration>
+```
+
+However, the NuGet Client plugins functionality is a global setting.
+This means that the plugins are discovered based on the platform and target framework, as discussed earlier, and there is currently no way to configure the dependent plugins via NuGet settings per repository.
+It could be a future possibility to provide users with an option to manage their plugins per repository instead of loading all the available plugins.
+However, given the limited number of plugin implementations currently available, this is not a problem that needs to be solved at this point in time, in my understanding.
+
+## Prior Art
+
+- The AWS team has developed a [.NET tool](https://docs.aws.amazon.com/codeartifact/latest/ug/nuget-cli.html#nuget-configure-cli) to simplify authentication with their private NuGet feeds.
+In their current implementation, they've added a subcommand, `codeartifact-creds install`.
+This command copies the credential provider to the NuGet plugins folder.
+They also provide an `uninstall` subcommand to remove the files from the NuGet plugin folders.
+However, due to limitations in the NuGet Client tooling, they've had to maintain plugins for both .NET Framework and .NET Core tooling. Additionally, they've had to provide commands to install and uninstall the NuGet plugins.
+
+## Unresolved Questions
+
+- The issue regarding workload installation on Mac and Linux, as well as NuGet credential providers, is discussed in [GitHub Issue #35912](https://github.com/dotnet/sdk/issues/35912#issuecomment-1759774310).
+When executing .NET SDK workload commands under sudo, the HOME directory path is modified, preventing the .NET SDK from writing files owned by root to the user's regular HOME directory.
+This alteration can cause problems when running standard commands without sudo and also interferes with NuGet credential providers.
+We have proposed a [workaround](https://github.com/dotnet/sdk/issues/35912#issuecomment-2004522180) for this issue.
+
+## Future Possibilities
+
+### Managing NuGet plugins per repository using .NET Local Tools
```

Comment:
It's worthwhile here mentioning the drawback that the tool must be executed via the `dotnet` CLI, since it's not on the PATH.

Reply:
I agree. Updated spec.